src/api/app/jobs/project_status_calculator.rb

```diff
@@ -211,7 +212,7 @@ def calc_status(opts = {})
 
     links = Array.new
     # find links
-    mypackages.values.each do |package|
+    mypackages.each_value.each do |package|
       if package.project == @dbproj.name and package.links_to_id
```

Comment:
and if you do each_value.each you create an array too and then an iterator on top of it